### DISCUSSION TITLE
List view keeps rebuilding on scroll

### DISCUSSION THREAD

**Discussion by kevinZartek** [rrousselGit/riverpod#1529] [General]
I have used the scoped provider logic and made the list item  const ( just like in marvel example), but on scroll the list keep rebuilding.
I am implementing a chat screen which has a auto translate feature, so the list item is watching a translation api provider (future provider ) which returns the translated string. Each time I scroll downwards or upwards the api call is translated.

**Comment by rrousselGit** [comment]
Hello!

Do you have more information? A reproducible example would be ideal. Or at least a short readable code snippet
Otherwise it's difficult to know what the problem is.


A rough guess would be that you're using `family` with a class as parameter, and that class does not override ==/hashCode.

**Reply by kevinmathew94** [reply]
*******
 Chat Screen ( Consumer Stateful Widget)

 @override
  Widget build(BuildContext context) {
    final chatState =
        ref.watch(businessChatNotifierProvider(widget.otherUser.id.toString()));
    final chatViewModel = ref.read(
        businessChatNotifierProvider(widget.otherUser.id.toString()).notifier);
.
.
.
chatState.when(
                      data: (chats) => chats.isEmpty
                          ? Center(
                              child: Text(
                              'No chats')):
 ListView.builder(
                              padding: EdgeInsets.zero,
                              shrinkWrap: true,
                              reverse: true,
                              controller: chatViewModel.scrollController,
                              itemCount: chats.length,
                              itemBuilder: (context, index) {
                                ChatModel chat = chats[index];
                                return ProviderScope(overrides: [
                                  currentChatMessage.overrideWithValue(chat),
                                ], child: const Message());
                              },
                            ),
 *******

*******
class Message extends ConsumerWidget {
  const Message({
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final message = ref.watch(currentChatMessage);
    String translated =
        ref.watch(translatedStringProvider(message.text ?? '')).when(
              loading: () => ('translating'),
              error: (err, stack) => (err.toString()),
              data: (translation) => (translation),
            );
   return .....

*******

final translatedStringProvider =
    FutureProvider.autoDispose.family<String, String>((ref, id) async {
  final cancelToken = CancelToken();
  ref.onDispose(cancelToken.cancel);
  final toLanguageCode = Translations.getLanguageCode('English');
  final repository = ref.watch(googleApiRepositoryProvider);
  final translatedString = await repository.fetchTranslation(
    id,
    toLanguageCode,
    cancelToken: cancelToken,
  );ref.maintainState;
  return translatedString;
});

*******

**Reply by kevinmathew94** [reply]
But issue is not with future provider for translation, because even though i comment the part where the item watches the provider for translated string, rebuilds are still there.
I think on scroll items are disposed and then called again on scroll back
